packages/apollo-gateway/src/__tests__/integration/networkRequests.test.ts

```diff
@@ -262,20 +262,23 @@ it('Rollsback to a previous schema when triggered', async () => {
   const onChange = jest.fn();
   const gateway = new ApolloGateway();
   await gateway.load({ engine: { apiKeyHash, graphId: serviceName } });
+
   gateway.onSchemaChange(onChange);
-  jest.runOnlyPendingTimers();
+
+  jest.advanceTimersByTime(10000);
 
   jest.useRealTimers();
-  await new Promise(resolve => setTimeout(resolve, 100));
+  await new Promise(resolve => setTimeout(resolve, 10));
```

Comment:
We tried this, but it strangely didn't work :/ I think when we did that 5 times it worked 😢

Reply:
In the operation registry plugin tests, we use a technique driven by `Promise`s, rather than tinkering with timers.  e.g.,:

```js
 await agent.requestPending();
```

Reply:
This didn't work for me. Though the pending driven approach seems ...`Promise`ing!